src/enclave/ServiceProvider/ServiceProvider.cpp

```
@@ -147,284 +151,197 @@ void ServiceProvider::export_public_key_code(const std::string &filename) {
-  if (!ias_client_cert_file) {
-    throw std::runtime_error(
-      "Set $IAS_CLIENT_CERT_FILE to the file generated by openssl x509, probably called "
-      "${OPAQUE_HOME}/client.crt.");
+  if (res != 0) {
+    printf("mbedtls_pk_parse_public_key failed with %d\n", res);
+    goto exit;
   }
-  ias->client_cert(ias_client_cert_file, "PEM");
-
-  const char *ias_client_cert_key_password_file = std::getenv("IAS_CLIENT_CERT_KEY_PASSWORD_FILE");
-  std::string password;
-  if (ias_client_cert_key_password_file) {
-    std::ifstream f(ias_client_cert_key_password_file);
-    std::stringstream s;
-    s << f.rdbuf();
-    password = s.str();
-  }
-  const char *ias_client_key_file = std::getenv("IAS_CLIENT_KEY_FILE");
-  if (!ias_client_key_file) {
-    throw std::runtime_error(
-      "Set $IAS_CLIENT_KEY_FILE to the file generated by openssl genrsa, probably called "
-      "${OPAQUE_HOME}/client.key.");
-  }
-  ias->client_key(ias_client_key_file, password.empty() ? nullptr : password.c_str());
 
-  ias->proxy_mode(IAS_PROXY_NONE);
-
-  ias->agent("wget");
+  pk_type = mbedtls_pk_get_type(&ctx);
+  if (pk_type != MBEDTLS_PK_RSA) {
+    printf("mbedtls_pk_get_type had incorrect type: %d\n", res);
+    goto exit;
+  }
 
-  X509 *signing_ca;
-  if (!cert_load_file(&signing_ca, ias_report_signing_ca_file.c_str())) {
-    throw std::runtime_error(std::string("Could not load ") + ias_report_signing_ca_file);
+  rsa_ctx = mbedtls_pk_rsa(ctx);
+  modulus_size = mbedtls_rsa_get_len(rsa_ctx);
+  modulus = (uint8_t*)malloc(modulus_size);
+  if (modulus == NULL) {
+    printf("malloc for modulus failed with size %zu:\n", modulus_size);
+    goto exit;
   }
-  X509_STORE *store = cert_init_ca(signing_ca);
-  if (!store) {
-    throw std::runtime_error("Could not initialize certificate store.");
+
+  res = mbedtls_rsa_export_raw(rsa_ctx, modulus, modulus_size, NULL, 0, NULL, 0, NULL, 0, NULL, 0);
+  if (res != 0) {
+    printf("mbedtls_rsa_export failed with %d\n", res);
```

Instead of printf can we instead throw a runtime error?